observer/src/filters/WorldFilter.cpp

```diff
@@ -129,4 +132,29 @@ const std::unique_ptr<BallFilter> &WorldFilter::bestFilter(const std::vector<std
     }
     return filters[bestIndex];
 }
-}  // namespace world
\ No newline at end of file
+
+void WorldFilter::process(std::vector<proto::SSL_DetectionFrame> visionFrames) {
+    std::sort(visionFrames.begin(),visionFrames.end(),[](const proto::SSL_DetectionFrame& a, const proto::SSL_DetectionFrame&b){
+        return a.t_capture() < b.t_capture();
+    });
+    for(const auto& frame : visionFrames){
+        addFrame(frame);
+    }
+    if(!visionFrames.empty()){
+        double latestTime = visionFrames.back().t_capture();
+        if(latestTime>lastUpdateTime){
+            lastUpdateTime = latestTime;
+        }
+    }
+    update(lastUpdateTime,false);
+}
+
+void WorldFilter::setRobotParameters(const TwoTeamRobotParameters& parameters) {
+//TODO: use robot parameters for collision detection etc.
```

Comment:
This sounds needed

Reply:
We don't do any collision detection yet. This is just function handles (I plan to change this in the future TM).